website/content/en/docs/upgrading/v1-migration.md

```diff
@@ -176,233 +289,297 @@ kubectl patch customresourcedefinitions ec2nodeclasses.karpenter.k8s.aws -p "{\"
+Prematurely removing the compatibility annotation will result in Node drift.
 
+If you need to rollback to a pre-`v1.0` version after removing the compatibility annotation, you must re-add it before rolling back.
 {{% /alert %}}
 
-3. Rollback the Karpenter Policy
-
-**v0.33 and v0.34:**
-```bash
-export TEMPOUT=$(mktemp)
-curl -fsSL https://raw.githubusercontent.com/aws/karpenter-provider-aws/"${KARPENTER_VERSION}"/website/content/en/preview/getting-started/getting-started-with-karpenter/cloudformation.yaml > ${TEMPOUT} \
-    && aws cloudformation deploy \
-    --stack-name "Karpenter-${CLUSTER_NAME}" \
-    --template-file "${TEMPOUT}" \
-    --capabilities CAPABILITY_NAMED_IAM \
-    --parameter-overrides "ClusterName=${CLUSTER_NAME}"
-```
-
-**v0.35+:**
-```bash
-export TEMPOUT=$(mktemp)
-curl -fsSL https://raw.githubusercontent.com/aws/karpenter-provider-aws/v"${KARPENTER_VERSION}"/website/content/en/preview/getting-started/getting-started-with-karpenter/cloudformation.yaml > ${TEMPOUT} \
-    && aws cloudformation deploy \
-    --stack-name "Karpenter-${CLUSTER_NAME}" \
-    --template-file "${TEMPOUT}" \
-    --capabilities CAPABILITY_NAMED_IAM \
-    --parameter-overrides "ClusterName=${CLUSTER_NAME}"
+If you have multiple `NodePools` that refer to the same `EC2NodeClass`, but have varying kubelet configurations, you will need to create a separate `EC2NodeClass` for unique set of kubelet configurations.
+
+For example, consider the following `v1beta1` manifests:
+```yaml
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: nodepool-a
+spec:
+  template:
+    spec:
+      kubelet:
+        maxPods: 10
+      nodeClassRef:
+        apiVersion: karpenter.k8s.aws/v1beta1
+        kind: EC2NodeClass
+        name: nodeclass
+---
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: nodepool-b
+spec:
+  template:
+    spec:
+      kubelet:
+        maxPods: 20
+      nodeClassRef:
+        apiVersion: karpenter.k8s.aws/v1beta1
+        kind: EC2NodeClass
+        name: nodeclass
+---
+apiVersion: karpenter.k8s.aws/v1beta1
+kind: EC2NodeClass
+metadata:
+  name: nodeclass
 ```
 
-4. Rollback the CRDs
-
-```bash
-helm upgrade --install karpenter-crd oci://public.ecr.aws/karpenter/karpenter-crd --version "${KARPENTER_VERSION}" --namespace "${KARPENTER_NAMESPACE}" --create-namespace \
-  --set webhook.enabled=true \
-  --set webhook.serviceName=karpenter \
-  --set webhook.port=8443
+In this example, we have two `NodePools` with different `kubelet` values, but they refer to the same `EC2NodeClass`.
+When handling conversion, Karpenter will annotate the `NodePools` with the `compatibility.karpenter.sh/v1beta1-kubelet-conversion` annotation.
+Before upgrading to `v1.1`, you must update your `NodePools` to refer to separate `EC2NodeClasses` to retain this behavior.
+
+```yaml
+apiVersion: karpenter.sh/v1
+kind: NodePool
+metadata:
+  name: nodepool-a
```

Comment:
IMO I think we should add what the kubelet compatibility annotation should be here. That way we can convay the order of moving the kubelet config to the EC2NodeClass would be:
1. Upgrade to 1.0.x
2. Add kubelet config to EC2NodeClass 
3. Remove the kubelet annotation

We need to emphasizes the importance of removing the annotation, as Karpenter won't respect what is set in the nodeclass if not dropped